src/Analyzer/Passes/QueryAnalysisPass.cpp

```diff
@@ -6887,13 +6887,12 @@ void QueryAnalyzer::resolveQuery(const QueryTreeNodePtr & query_node, Identifier
                 scope.scope_node->formatASTForErrorMessage());
     }
 
-    std::erase_if(with_nodes, [](const QueryTreeNodePtr & node)
-    {
-        auto * subquery_node = node->as<QueryNode>();
-        auto * union_node = node->as<UnionNode>();
-
-        return (subquery_node && subquery_node->isCTE()) || (union_node && union_node->isCTE());
-    });
+    /** WITH section can be safely removed, because WITH section only can provide aliases to query expressions
```

Comment:
And everything mentioned already resolved? CTEs registered above and what about regular subqueries with  aliases or constants?

So, we will not have WITH sections after the pass, constants, cte and aliased sub-queries will be substituted?

Reply:
No, something inside of WITH clause can be unresolved at that point but all the CTEs are registered in the query scope, so it'll be resolved on the first usage.